src/modules/commander/state_machine_helper.cpp

```diff
@@ -1076,8 +1076,12 @@ void set_link_loss_nav_state(vehicle_status_s *status,
 	} else if (link_loss_act == link_loss_actions_t::AUTO_RTL
 		   && status_flags->condition_global_position_valid && status_flags->condition_home_position_valid) {
 
-		uint8_t main_state_prev = 0;
-		main_state_transition(status, commander_state_s::MAIN_STATE_AUTO_RTL, main_state_prev, status_flags, internal_state);
+		int32_t link_regained_action = 1;
+		param_get(param_find("COM_LINK_REG_ACT"), &link_regained_action);
```

Comment:
We shouldn't access params like this in here. There's some overhead to param_get.

I would add it as a BlockParam in commander (https://github.com/PX4/Firmware/blob/master/src/modules/commander/Commander.hpp#L95) and pass it through.